lib/remoteproc/remoteproc_virtio.c

```diff
@@ -309,3 +309,14 @@ int rproc_virtio_notified(struct virtio_device *vdev, uint32_t notifyid)
 	}
 	return 0;
 }
+
+void rproc_virtio_wait_remote_ready(struct virtio_device *vdev)
+{
+	uint8_t status;
+
+	while (1) {
+		status = rproc_virtio_get_status(vdev);
```

Comment:
This checking is only available to virtio slave, please add "if (vdev->role == VIRTIO_DEV_SLAVE)"
if it is MASTER, for not, always return.
There can be future improvement to allow non-busy-looping wait, we can define default poll in libmetal for the platform, which will do nothing, and allow application to overwrite it, and in this wait function, we call the platform poll function.

Reply:
Concerning the test: this was my first implementation. But rproc_virtio_get_status is available for the slave and the master. So it could be available for both (no use case in mind for a usage on master side). 
Please tell me your preference.

Concerning the wait: not sure to understand... do you suggest to not implement the loop here but only provide access to the status? in this case an alternative could be to export the rproc_virtio_get_status function instead of implementing this new service.

Reply:
I should have put my comments one line after, the patch is checking status bit DRIVER_OK which will be set by master, and now, slave is not able to tell its status unless we update the rsc table.
About waiting, for now, it is busy wait, but it can be future improved to call platform wait if the status bit is not set